api/types/swarm/config.go

```diff
@@ -0,0 +1,31 @@
+package swarm
+
+import "os"
+
+// Config represents a config.
+type Config struct {
+	ID string
+	Meta
+	Spec ConfigSpec
+}
+
+// ConfigSpec represents a config specification from a config in swarm
+type ConfigSpec struct {
+	Annotations
+	Data []byte `json:",omitempty"`
+}
+
+// ConfigReferenceFileTarget is a file target in a config reference
+type ConfigReferenceFileTarget struct {
+	Name string
```

Comment:
It's called `Name` for consistency with secrets. With #32571, secrets will also allow this to contain an absolute path. Changing the name in `SecretReferenceFileTarget` would be a breaking API change, unfortunately.